pkg/services/ldap/ldap.go

```diff
@@ -148,10 +154,78 @@ func (server *Server) Login(query *models.LoginUserQuery) (
 	return user, nil
 }
 
+// getUsersIteration is a helper function for Users() method.
+// It divides the users by equal parts for the anticipated requests
+func getUsersIteration(logins []string, fn func(int, int) error) error {
+	lenLogins := len(logins)
+	iterations := int(
+		math.Ceil(
+			float64(lenLogins) / float64(UsersMaxRequest),
+		),
+	)
+
+	for i := 1; i < iterations+1; i++ {
```

Comment:
I think we could simplify this loop to something like this:

```go
	for i := 1; i < iterations+1; i++ {
		previous := math.Max(0, float64(UsersMaxRequest*(i-1)))
		current := math.Min(float64(i*UsersMaxRequest), float64(lenLogins))

		err := fn(int(previous), int(current))
		if err != nil {
			return err
		}
	}
```

Reply:
ooh, I love it!

Reply:
Managed to simplified it even further :)

Reply:
Neat!